contracts/Manager.sol

```diff
@@ -333,364 +214,190 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
-            if (getElection().canReceiveVotes(group, votes + scheduledVotes)) {
-                votableGroups[numberVotableGroups] = group;
-                numberVotableGroups++;
-            }
+    function changeStrategy(address newStrategy) public {
+        if (
+            newStrategy != address(0) &&
+            (!specificGroupStrategy.isSpecificGroupStrategy(newStrategy) ||
+                !groupHealth.isValidGroup(newStrategy))
+        ) {
+            revert GroupNotEligible(newStrategy);
         }
 
-        address[] memory votableGroupsFinal = new address[](numberVotableGroups);
-        for (uint256 i = 0; i < numberVotableGroups; i++) {
-            votableGroupsFinal[i] = votableGroups[i];
+        uint256 stCeloAmount = stakedCelo.balanceOf(msg.sender);
+        if (stCeloAmount != 0) {
+            address currentStrategy = strategies[msg.sender];
+            _transfer(currentStrategy, newStrategy, stCeloAmount, msg.sender, msg.sender);
         }
 
-        return votableGroupsFinal;
+        strategies[msg.sender] = _checkStrategy(newStrategy);
     }
 
     /**
-     * @notice Sorts an array of GroupWithVotes structs based on increasing
-     * `votes` values.
-     * @param groupsWithVotes The array to sort.
-     * @dev This is an in-place insertion sort. In general in Solidity we should
-     * be careful of algorithms on arrays, especially O(n^2) ones, but here
-     * we're guaranteed to be working with a small array, its length is bounded
-     * by the maximum number of groups that can be voted for in Elections.sol.
-     */
-    function sortGroupsWithVotes(GroupWithVotes[] memory groupsWithVotes) internal pure {
-        for (uint256 i = 1; i < groupsWithVotes.length; i++) {
-            uint256 j = i;
-            while (j > 0 && groupsWithVotes[j].votes < groupsWithVotes[j - 1].votes) {
-                (groupsWithVotes[j], groupsWithVotes[j - 1]) = (
-                    groupsWithVotes[j - 1],
-                    groupsWithVotes[j]
-                );
-                j--;
-            }
-        }
+     * @notice Unlock vote balance of stCelo.
+     * @param accountAddress The account to be unlocked.
+     */
+    function unlockBalance(address accountAddress) public {
+        stakedCelo.unlockVoteBalance(accountAddress);
+    }
+
+    /**
+     * @notice Rebalances Celo between groups that have incorrect Celo-stCelo ratio.
+     * FromGroup is required to have more Celo than it should and ToGroup needs
+     * to have less Celo than it should.
+     * @param fromGroup The from group.
+     * @param toGroup The to group.
+     */
+    function rebalance(address fromGroup, address toGroup) public {
+        address[] memory fromGroups;
+        address[] memory toGroups;
+        uint256[] memory fromVotes;
+        uint256[] memory toVotes;
+        (fromGroups, toGroups, fromVotes, toVotes) = groupHealth.rebalance(fromGroup, toGroup);
```

Comment:
Is it necessary due to bytecode size constraints for this logic to live outside of Manager? Personally feel it'd be nice if GroupHealth was just limited to checking group health, other features like this one fit inside Manager better imo.

Reply:
It was the original reason but after lates refactor I was able to return it back to Manager